src/main/java/catobot/Storage.java

```diff
@@ -0,0 +1,119 @@
+package catobot;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import catobot.exception.EmptyCommandException;
+import catobot.exception.LoadingException;
+import catobot.item.Deadline;
+import catobot.item.Event;
+import catobot.item.Task;
+import catobot.item.Todo;
+
+/**
+ * Represents the local memory of tasks.
+ */
+public class Storage {
+    /** The path to the file. */
+    private final String filePath;
+
+    /** The file storing the tasks. */
+    private final File storage;
+
+    /**
+     * Constructor for Storage.
+     *
+     * @param filePath The path to the local memory.
+     * @throws LoadingException If the file cannot be loaded properly.
+     */
+    public Storage(String filePath) throws LoadingException {
+        this.filePath = filePath;
+        this.storage = new File(filePath);
+        try {
+            if (!storage.exists()) {
+                storage.getParentFile().mkdirs();
+                storage.createNewFile();
+            }
+        } catch (IOException e) {
+            throw new LoadingException();
+        }
+    }
+
+    /**
+     * Loads the tasks from local memory to Catobot.
+     *
+     * @return The list of tasks from local memory.
+     * @throws LoadingException If the tasks cannot be loaded from storage.
+     */
+    public List<Task> load() throws LoadingException {
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        try {
+            Scanner s = new Scanner(storage);
+
+            while (s.hasNext()) {
+                String rawTask = s.nextLine();
+                String[] input = rawTask.split(" \\| ", 4);
+
+                loadTask(input, taskList);
+            }
```

Comment:
Could we further abstract this into another method ? seems a couple of things is happening to the load() ?